fendermint/vm/topdown/src/convert.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! Handles the type conversion to ethers contract types
+
+use crate::IPCParentFinality;
+use anyhow::anyhow;
+use ethers::abi::{Function, Token};
+use ethers::types::Bytes;
+use ethers::types::U256;
+use fendermint_vm_ipc_actors::{gateway_getter_facet, gateway_router_facet};
+use fvm_shared::address::{Address, Payload};
+use ipc_agent_sdk::message::ipc::ValidatorSet;
+use std::str::FromStr;
+
+const COMMIT_PARENT_FINALITY_FUNC_NAME: &str = "commitParentFinality";
+const GET_LATEST_PARENT_FINALITY_FUNC_NAME: &str = "getLatestParentFinality";
+
+impl TryFrom<IPCParentFinality> for gateway_router_facet::ParentFinality {
+    type Error = anyhow::Error;
+
+    fn try_from(value: IPCParentFinality) -> Result<Self, Self::Error> {
+        if value.block_hash.len() != 32 {
+            return Err(anyhow!("invalid block hash length, expecting 32"));
+        }
+
+        let mut array = [0u8; 32];
```

Comment:
Minor but if you call this `block_hash` then it carries the meaning, it still does not shadow `value.block_hash`, and you don't have to type `block_hash: array` when assigning in `Self`, just its name.